File: java/src/test/java/com/qmates/kata/extractinterface/FakeChargePaymentGateway.java
package com.qmates.kata.extractinterface;

public class FakeChargePaymentGateway implements ChargePaymentGateway {
  private final ChargeResult fixedResult;

  public FakeChargePaymentGateway(ChargeResult fixedResult) {
    this.fixedResult = fixedResult;
  }

  @Override
  public ChargeResult charge(long amountCents, String paymentToken) {
    return this.fixedResult;
  }

}

File: typescript/src/01-extract-interface/fake-charge-payment-gateway.ts
import type { ChargeResult } from "./stripe-payment-gateway";
import type { Cents } from "../domain";
import type ChargePaymentGateway from "./charge-payment-gateway";

export default class FakeChargePaymentGateway implements ChargePaymentGateway {
  private readonly fixedChargeResult: ChargeResult;

  constructor(fixedChargeResult: ChargeResult) {
    this.fixedChargeResult = fixedChargeResult;
  }

  charge(amountCents: Cents, paymentToken: string): Promise<ChargeResult> {
    return Promise.resolve(this.fixedChargeResult);
  }
}
